### PR TITLE
Add type search

### PR DESCRIPTION
**Pull request by ashu8912**
# [Title: describe the change in one sentence]
Allow to search by type and add facility of searching by other properties as well

## How to use
Go to any view which has a type column try using the search filter and you now should be able to search by type as well
## Testing done
Try using the search filter on a view that has a type column and you will see you are now able to search by type and namespace both

### CODE REVIEW

File: frontend/src/components/role/BindingList.tsx [hunk 1]
```diff
@@ -70,6 +70,7 @@ export default function RoleBindingList() {
       <SimpleTable
         rowsPerPage={[15, 25, 50]}
         filterFunction={filterFunc}
+        advancedFilters={['kind']}
```
Comment (joaquimrocha) [inline frontend/src/components/role/BindingList.tsx:73]: We should add it also to the Workload overview list, as that'll mix a lot of different resource types.

**Review by joaquimrocha — CHANGES_REQUESTED** [review]

File: frontend/src/components/common/SimpleTable.tsx [hunk 2]
```diff
@@ -234,6 +240,11 @@ export default function SimpleTable(props: SimpleTableProps) {
 
   if (filterFunction) {
     filteredData = displayData.filter(filterFunction);
+    for (let i = 0; advancedFilters && i < advancedFilters.length; i++) {
+      filteredData = displayData.filter((item: KubeObjectInterface) =>
```
[the remark this answers is not in the document]
Comment (ashu8912) [reply frontend/src/components/common/SimpleTable.tsx]: Fixed

File: frontend/src/components/common/SimpleTable.tsx [hunk 3]
```diff
@@ -234,6 +240,11 @@ export default function SimpleTable(props: SimpleTableProps) {
 
   if (filterFunction) {
     filteredData = displayData.filter(filterFunction);
+    for (let i = 0; advancedFilters && i < advancedFilters.length; i++) {
```
[the remark this answers is not in the document]
Comment (ashu8912) [reply frontend/src/components/common/SimpleTable.tsx]: done

File: frontend/src/components/role/BindingList.tsx (hunk 1 above)
Comment (ashu8912) [reply frontend/src/components/role/BindingList.tsx:73]: done

**Review by joaquimrocha — APPROVED** [review]

**Pull request merged**